src/Razor/src/rzls/Program.cs

```diff
@@ -102,42 +103,49 @@ public static async Task Main(string[] args)
 
         loggerFactory.GetOrCreateLogger("RZLS").LogInformation($"Razor Language Server started successfully.");
 
-        try
-        {
-            await host.WaitForExitAsync().ConfigureAwait(true);
-        }
-        finally
-        {
-            devKitTelemetryReporter?.Flush();
-        }
+        await host.WaitForExitAsync().ConfigureAwait(true);
     }
 
-    private static async Task<ITelemetryReporter?> TryGetTelemetryReporterAsync(string telemetryLevel, string sessionId, string telemetryExtensionPath)
+    private static async Task<TelemetryContext?> TryGetTelemetryReporterAsync(string telemetryLevel, string sessionId, string telemetryExtensionPath)
     {
-        ITelemetryReporter? devKitTelemetryReporter = null;
+        ExportProvider? exportProvider = null;
         if (!telemetryExtensionPath.IsNullOrEmpty())
         {
             try
             {
-                using var exportProvider = await ExportProviderBuilder
+                exportProvider = await ExportProviderBuilder
                     .CreateExportProviderAsync(telemetryExtensionPath)
                     .ConfigureAwait(true);
 
                 // Initialize the telemetry reporter if available
-                devKitTelemetryReporter = exportProvider.GetExports<ITelemetryReporter>().SingleOrDefault()?.Value;
+                var devKitTelemetryReporter = exportProvider.GetExports<ITelemetryReporter>().SingleOrDefault()?.Value;
 
                 if (devKitTelemetryReporter is ITelemetryReporterInitializer initializer)
                 {
                     initializer.InitializeSession(telemetryLevel, sessionId, isDefaultSession: true);
+                    return new TelemetryContext(exportProvider, devKitTelemetryReporter);
+                }
+                else
+                {
+                    exportProvider.Dispose();
                 }
             }
             catch (Exception ex)
             {
                 await Console.Error.WriteLineAsync($"Failed to load telemetry extension in {telemetryExtensionPath}.").ConfigureAwait(true);
                 await Console.Error.WriteLineAsync(ex.ToString()).ConfigureAwait(true);
+                exportProvider?.Dispose();
             }
         }
 
-        return devKitTelemetryReporter;
+        return null;
+    }
+
+    private record struct TelemetryContext(IDisposable ExportProvider, ITelemetryReporter TelemetryReporter) : IDisposable
```

Comment:
Consider making this a readonly record struct so that the `ExportProvider` and `TelemetryReporter` properties aren't mutable.

```suggestion
    private readonly record struct TelemetryContext(IDisposable ExportProvider, ITelemetryReporter TelemetryReporter) : IDisposable
```